spec/unit/shoulda/matchers/active_model/validate_presence_of_matcher_spec.rb

```diff
@@ -37,6 +37,27 @@
     )
   end
 
+  context 'a model with a private setter method' do
+    let(:record) do
```

Comment:
The convention in these test files is to use builder methods instead of `let`s. Namely, we often have a pair of methods: `define_model_...`, which defines and returns a model class, and `build_record_...`, which returns an instance of that model. See the bottom of this file for an example of the `create_model` / `build_record` pattern (although [the tests for the numericality matcher](https://github.com/thoughtbot/shoulda-matchers/blob/master/spec/unit/shoulda/matchers/active_model/validate_numericality_of_matcher_spec.rb#L1789) might be a slightly better example, since the methods are actually named consistently there).

There are a few different helper methods you can use to make models in tests, but the one that's applicable here is called `define_active_model_class`, and you can see an example of how it's used in this file on [line 328](https://github.com/thoughtbot/shoulda-matchers/blob/master/spec/unit/shoulda/matchers/active_model/validate_presence_of_matcher_spec.rb#L328).

With these two methods in place, then you can write something like:

``` ruby
it 'will not throw an exception' do
  # here we are assuming you've made a method called `build_record_with_private_attribute`
  record = build_record_with_private_attribute
  assertion = -> { expect(record).to matcher }
  expect(&assertion).not_to raise_error
end
```